js/kucoin2.js

```diff
@@ -265,6 +265,26 @@ module.exports = class kucoin2 extends Exchange {
         return result;
     }
 
+    async fetchFundingFees (codes = undefined, params = {}) {
+        const numberOfCodes = codes.length;
+        if (numberOfCodes !== 1) {
+            throw new ArgumentsRequired (this.id + ' fetchFundingFees supports only one currency per call');
```

Comment:
I suggest that we don't throw from here and instead use the approach implemented in `buda` or `bibox`. Can you rewrite it a bit?

Reply:
In hindsight I think we designed this method wrong. In python if I need to make 100 requests I need to call `asyncio.gather` on the each coroutine. If we await for more than one request in a single coroutine I cannot write it asyncly in python.

Reply:
The issue is sometimes they provide an endpoint for all funding fees, and sometimes you need to specify it separately. Also the bibox `fetchFundingFees` has been broken for like 8 months

Reply:
> In hindsight I think we designed this method wrong.

For one currency, we just need another one, called `fetchFundingFee(code)` (for one currency).

So, yes, the better design would be to have either this or that method, depending on what is provided by the exchange. I would not leave it this way, however, i'd suggest to rewrite the places where you think it is designed badly. But I would not do this for all exchanges at once, instead I'd rather do one exchange at a time. We could start from KuCoin in particular. What do you think?

Reply:
I think this is ok, until we get an exchange like yobit's `fetchTickers` comes along which that lets you specify multiple symbols in the url, then we screwed. Plus having two methods for this kinda ununifies it, so I suggest we keep the signature the same and just throw an error if people try to get too many codes at once.

Reply:
Also in python the is a whole callchain that doesn't exist in other languages called `loadFees`  and this relies on `fetchFundingFees` to return that exact structure (so it can `this.extend` `this.fees['funding`])

Reply:
I think you are right with regard to the naming, it is important we keep the structure returned from `fetchFundingFee` like I defined it, so that it will be compatible with `fetchFundingFees` and `this.fees['funding']`

Reply:
> and just throw an error

Can't tolerate it, sorry. Will have to rewrite it then, throwing an error kinda ununifies it as well.

> Also in python there is a whole callchain that doesn't exist in other languages called

Yes, this has to be properly designed and implemented in all three languages, taking into these details:

- https://github.com/ccxt/ccxt/pull/1669#issuecomment-362535401
- https://github.com/ccxt/ccxt/issues/4128#issuecomment-437005457
- https://github.com/ccxt/ccxt/pull/1835#issuecomment-368326638

Current call chains are just not flexible enough to cover the entire variety. So we will have to extend this aspect to two methods most likely, otherwise it doesn't really scale.